### ISSUE TITLE
Update rails with --conservative

### ISSUE THREAD

**Issue by ylecuyer** [rubygems/rubygems#5559]
### Describe the problem as clearly as you can

On our team we instructed people to use `--conservative` when upgrading gem to avoid unexpected upgrades. When upgrading rails however, as it is a bundling gem installing other gems, the fail to get the expected result with `bundle update --conservative rails`.

The proper command should be `bundle update --conservative rails activesupport actionpack actionview activemodel activerecord actionmailer activejob actioncable activestorage actionmailbox actiontext railties`

### What were you expecting to happen?

I would expect `bundle update --conservative rails` to upgrade also dependent gems.

### What actually happened?

I get the following error:

```
bundle update --conservative rails
Fetching gem metadata from https://rubygems.org/.......
Resolving dependencies................
Bundler could not find compatible versions for gem "actionmailer":
  In snapshot (Gemfile.lock):
    actionmailer (= 6.1.4.7)

  In Gemfile:
    devise_invitable was resolved to 2.0.2, which depends on
      actionmailer (>= 5.0)

    rails (~> 7) was resolved to 7.0.0, which depends on
      actionmailer (= 7.0.0)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

Whereas this one works:

```
bundle update --conservative rails activesupport actionpack actionview activemodel activerecord actionmailer activejob actioncable activestorage actionmailbox actiontext railties
Fetching gem metadata from https://rubygems.org/.......
Resolving dependencies........................................
[SKIPED]
Bundle updated!
````

I don't know how we can achieve this though, should we add a `bundle` attribute to gems? or check gems with no ruby files in the `files`? or just make a hard coded rule for rails (and other gems with this pattern)?

**Comment by deivid-rodriguez** [comment]
I think this should be fixed by #5520!

**Comment by ylecuyer** [comment]
> I think this should be fixed by #5520!

Indeed, thanks for the information, I'll close in favor of #5520 then

**Comment by deivid-rodriguez** [comment]
Well, while I initially thought the same change would fix this, #5250 would need further tweaks for this to work, so I'll reopen.

**Comment by deivid-rodriguez** [comment]
This will be fixed by #5559!
